src/lib.rs

```diff
@@ -43,6 +43,14 @@ pub struct Cfg {
     pub listener: Listener,
 }
 
+// MessengerState tells us whether our onion messenger is still starting up is ready to start
+// forwarding messages.
+#[derive(Debug)]
+pub enum MessengerState {
```

Comment:
Rather than add a whole state machine, can we have a one-shot channel and close it when we're done with startup? Also comes with the benefit that we can select on shutdown / channel close and then we don't have to have any X second / magic number wait -> it'll either exit if we've given up or yield exactly when the onion messenger has started up.

Reply:
Yeah that sounds good. I changed the MessengerState to a channel, though I couldn't quite get a oneshot channel to work because it requires a "self" method receiver, which doesn't work for us. So I used a normal mpsc channel instead.